From f18e32c5affad9bf376103a536c7ded19366d92f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lubom=C3=ADr=20Sedl=C3=A1=C5=99?= <lsedlar@redhat.com>
Date: Mon, 4 Apr 2016 14:24:18 +0200
Subject: [PATCH 2/3] [atomic] Stop creating the os directory
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The output directory for Lorax can not exist beforehand, or the process
will fail.

Signed-off-by: Lubomír Sedlář <lsedlar@redhat.com>
---
 pungi/phases/atomic_installer.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pungi/phases/atomic_installer.py b/pungi/phases/atomic_installer.py
index 476ad1c..3390a74 100644
--- a/pungi/phases/atomic_installer.py
+++ b/pungi/phases/atomic_installer.py
@@ -49,7 +49,8 @@ class AtomicInstallerThread(WorkerThread):
         self.logdir = compose.paths.log.topdir('{}/atomic'.format(arch))
 
         source_variant = compose.variants[config['source_repo_from']]
-        source_repo = translate_path(compose, compose.paths.compose.repository(arch, source_variant))
+        source_repo = translate_path(
+            compose, compose.paths.compose.repository(arch, source_variant, create_dir=False))
 
         self._run_atomic_cmd(compose, variant, arch, config, source_repo)
 
@@ -100,7 +101,7 @@ class AtomicInstallerThread(WorkerThread):
         compose.im.add(variant.uid, arch, img)
 
     def _run_atomic_cmd(self, compose, variant, arch, config, source_repo):
-        image_dir = compose.paths.compose.os_tree(arch, variant)
+        image_dir = compose.paths.compose.os_tree(arch, variant, create_dir=False)
         lorax_wrapper = lorax.LoraxWrapper()
         cmd = lorax_wrapper.get_lorax_cmd(
             compose.conf['release_name'],
-- 
2.7.3

